tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -157,4 +177,101 @@ TEST(SGSparseMatrix, io_libsvm)
 			EXPECT_NEAR(m[i].features[j].entry, m_from_file[i].features[j].entry, 1E-14);
 		}
 	}
+
+    //remove file generated by test
+	unlink("SGSparseMatrix_io_libsvm_output.txt");
+}
+
+TEST(SGSparseMatrix, access_by_index_non_square)
+{
+    //GeneratSparseMatrix function cannot used in this test since it uses () operator
+    CRandom randGenerator(0);
+    float64_t sparseLevel=0.1;
+    const index_t numberOfFeatures=50;
+    const index_t numberOfVectors=100;
+
+    SGSparseMatrix<float64_t> matrix(numberOfFeatures, numberOfVectors);
+
+    for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
+        for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
+        {
+            float64_t randomNumber=randGenerator.random(0.0,1.0);
+            if (randomNumber<=sparseLevel)
+                matrix(vecIndex,featIndex)=randomNumber;
```

Comment:
Can you call here `GenerateMatrix` as well so that this test only contains the `EXPECT_EQ` in the loops?